corehq/apps/export/models/new.py

```diff
@@ -587,10 +587,19 @@ def _create_index(_path, _transform):
 
         try:
             index = self._string_column_paths.index(string_item_path)
+            column = self.columns[index]
+            # on rare occasions, the paths may not match,
+            # and it's a sign to regenerate the cache
+            if column.item.path != item_path:
+                self._string_column_paths = [
+                    _create_index(column.item.path, column.item.transform)
+                    for column in self.columns
+                ]
+                index = self._string_column_paths.index(string_item_path)
+                column = self.columns[index]
```

Comment:
Does this need to be in side the `except ValueError` block? Previously `column = self.columns[index]` was not...

Oh, I think I see why, it's because of the second
```py
index = self._string_column_paths.index(string_item_path)
```
(line 598) I think it makes sense to move line 599 back to where it was below (just outside the try/except block). Do you agree?

Reply:
@millerdev I'm a little confused...this is a new line and a new condition (the original line is still there). Might be good to look at the chunk of code as a whole? I included everything in the same try, except block because there are now two possible conditions where the `ValueError` might get triggered, and the second condition is dependent on `self.columns[index]` being called before it. also the second call to `self.columns[index]` is dependent on the first call returning `None`.

This definitely feels a little hacky to me, but the tests were not passing prior to this change. I'm a little concerned now and feel that this needs to go through additional QA before merging in.

Reply:
i expected this release was as simple as removing the feature flag, but due to the issues with the tests...it turns out that's not the case

Reply:
To clarify on your original question, line 599 is inside a conditional. `if column.item.path != item_path:` so I don't think moving it outside of the `try, except` block makes too much sense as it will always be called twice (see line 590) rather than when the condition in line 593 is met.

Reply:
@millerdev ah ok got it